lib/ib/messages/incoming/ticks.rb

```diff
@@ -135,7 +135,119 @@ def valid?
           end
 
       TickSnapshotEnd = def_message 57, [:ticker_id, :int]
+=begin
+    def processTickByTickMsg(self, fields):
+        next(fields)
+        reqId = decode(int, fields)
+        tickType = decode(int, fields)
+        time = decode(int, fields)
 
+        if tickType == 0:
+            # None
+            pass
+        elif tickType == 1 or tickType == 2:
+            # Last or AllLast
+            price = decode(float, fields)
+            size = decode(int, fields)
+            mask = decode(int, fields)
+class TickAttribLast(Object):
+    def __init__(self):
+        self.pastLimit = False
+        self.unreported = False
+
+    def __str__(self):
+        return "PastLimit: %d, Unreported: %d" % (self.pastLimit, self.unreported)
+
+            tickAttribLast = TickAttribLast()
+            tickAttribLast.pastLimit = mask & 1 != 0
+            tickAttribLast.unreported = mask & 2 != 0
+            exchange = decode(str, fields)
+            specialConditions = decode(str, fields)
+
+            self.wrapper.tickByTickAllLast(reqId, tickType, time, price, size, tickAttribLast,
+                                           exchange, specialConditions)
+        elif tickType == 3:
+            # BidAsk
+            bidPrice = decode(float, fields)
+            askPrice = decode(float, fields)
+            bidSize = decode(int, fields)
+            askSize = decode(int, fields)
+            mask = decode(int, fields)
+	class TickAttribBidAsk(Object):
+    def __init__(self):
+        self.bidPastLow = False
+        self.askPastHigh = False
+
+    def __str__(self):
+        return "BidPastLow: %d, AskPastHigh: %d" % (self.bidPastLow, self.askPastHigh)
+
+
+            tickAttribBidAsk = TickAttribBidAsk()
+            tickAttribBidAsk.bidPastLow = mask & 1 != 0
+            tickAttribBidAsk.askPastHigh = mask & 2 != 0
+
+            self.wrapper.tickByTickBidAsk(reqId, time, bidPrice, askPrice, bidSize,
+                                          askSize, tickAttribBidAsk)
+        elif tickType == 4:
+            # MidPoint
+            midPoint = decode(float, fields)
+
+            self.wrapper.tickByTickMidPoint(reqId, time, midPoint)
+=end
+			TickByTick =  def_message 99, [:ticker_id, :int ],
+			[ :tick_type, :int],
+			[ :time, :int_date ]
+
```

Comment:
I had to declare here the correspnding fields, otherwise I was getting a method missing exception:

```ruby
[:size, :int],
[:price, :decimal],
[:exchange, :string],
[:bid_size, :int],
[:ask_size, :int],
[:bid_price, :decimal],
[:ask_price, :decimal],
[:mid_point, :decimal],
[:mask, :int]
```